subprojects/core/src/main/java/org/gradle/initialization/StartParameterBuildOptions.java

```diff
@@ -408,4 +413,78 @@ public void applyTo(StartParameterInternal settings, Origin origin) {
             settings.setExportKeys(true);
         }
     }
+
+    public static class ConfigurationCacheOption extends BooleanBuildOption<StartParameterInternal> {
+
+        public static final String PROPERTY_NAME = "org.gradle.unsafe.configuration-cache";
+        public static final String LONG_OPTION = "configuration-cache";
+
+        public ConfigurationCacheOption() {
+            super(PROPERTY_NAME, BooleanCommandLineOptionConfiguration.create(
+                LONG_OPTION,
+                "Enables the configuration cache. Gradle will try to reuse the build configuration from previous builds.",
+                "Disables the configuration cache."
+            ).incubating());
+        }
+
+        @Override
+        public void applyTo(boolean value, StartParameterInternal settings, Origin origin) {
+            settings.setConfigurationCacheEnabled(value);
+        }
+    }
+
+    public static class ConfigurationCacheFailOnProblemsOption extends BooleanBuildOption<StartParameterInternal> {
+
+        public static final String PROPERTY_NAME = "org.gradle.unsafe.configuration-cache.fail-on-problems";
+        public static final String LONG_OPTION = "configuration-cache-fail-on-problems";
```

Comment:
This could be changed to `--fail-on-configuration-cache-problems`
Or the other way around to `--ignore-configuration-cache-problems`

Reply:
👍 `--ignore-configuration-cache-problems`

Reply:
I think about evolving this towards the same semantics as `--warn-mode` so we can also evolve towards configuration cache problems and deprecations towards consistent behaviours (but a different default).